File: luiz/ba1c.py
#! /usr/bin/env python


TRANSLATION = {"A": "T", "C": "G", "G": "C", "T": "A"}


def revc(pattern):
    return "".join(TRANSLATION[c] for c in reversed(pattern))


assert revc("AAAACCCGGT") == "ACCGGGTTTT"


if __name__ == "__main__":
    with open("data/rosalind_ba1c.txt", "r") as dataset:
        text = dataset.readline().rstrip()
    print(revc(text))

File: luiz/src/bin/ba1c.rs
use std::env;
use std::error::Error;
use std::fs;

use bioinformatics_algorithms::revc;

fn main() -> Result<(), Box<dyn Error>> {
    let input: String = env::args()
        .nth(1)
        .unwrap_or("data/rosalind_ba1c.txt".into());
    let data = fs::read_to_string(input)?;
    let mut lines = data.lines();

    let text = lines.next().unwrap();

    println!("{}", revc(text));

    Ok(())
}
